src/lib/Hydra/Controller/User.pm

```diff
@@ -226,6 +272,29 @@ sub github_redirect :Path('/github-redirect') Args(0) {
     $c->res->redirect("https://github.com/login/oauth/authorize?client_id=$client_id&scope=user:email");
 }
 
+sub oidc_redirect :Path('/oidc-redirect') Args(0) {
+
+    my ($self, $c) = @_;
+
+    my $after = "/" . $c->req->params->{after};
+
+    $c->res->cookies->{'after_oidc'} = {
+        name => 'after_oidc',
+        value => $after,
+    };
+
+    my $oidc_client = get_oidc_client($c);
+    my $state = UUID4::Tiny::create_uuid_string();
+    $c->session->{oidc_state} = $state;
+    my $redirect_url = $oidc_client->uri_to_redirect(
+        redirect_uri => $c->uri_for('/oidc-login'),
+        scope        => q{openid},
```

Comment:
Would be nice to have this configurable

Reply:
I'm not sure that's desirable: we care about the `name`, `email` and `sub` fields which are standard in openid, and I would very much like to avoid going further than that and requiring mapping logic.